pkg/webui/console/store/middleware/logics/events.js

```diff
@@ -68,31 +67,42 @@ const createEventsConnectLogics = function (
         const connecting = status === CONNECTION_STATUS.CONNECTING
         if (connected || connecting) {
           reject()
+          return
         }
 
         allow(action)
       },
-      async process ({ action }, dispatch, done) {
+      async process ({ getState, action }, dispatch, done) {
         const { id } = action
 
         try {
           channel = await onEventsStart([ id ])
+
           channel.on('start', () => dispatch(startEventsSuccess(id)))
           channel.on('event', message => dispatch(getEventSuccess(id, message)))
-          channel.on('error', error => dispatch(getEventFailure(id, error)))
+          channel.on('error', function (error) {
+            const status = selectEntityEventsStatus(getState(), id)
+            if (status === CONNECTION_STATUS.CONNECTING) {
```

Comment:
see https://github.com/TheThingsNetwork/lorawan-stack/pull/974/files#r304372739, if you decide to go with option 3, then it would be nice to move this to selectors, something like `selectEventsStreamConnected`, etc.